lib/internal/process/write-coverage.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+const process = require('process');
+const path = require('path');
+const fs = require('fs');
+const mkdirSync = fs.mkdirSync;
+const writeFileSync = fs.writeFileSync;
+
+var isWritingCoverage = false;
+function writeCoverage() {
+  if (isWritingCoverage || !global.__coverage__) {
+    return;
+  }
+  isWritingCoverage = true;
+
+  const dirname = path.join(path.dirname(process.execPath), '.coverage');
+  const filename = `coverage-${process.pid}-${Date.now()}.json`;
+  try {
+    mkdirSync(dirname);
+  } catch (err) {
+    if (err.code !== 'EEXIST') {
+      console.error(err);
+      return;
+    }
+  }
+
+  const target = path.join(dirname, filename);
+  const coverageInfo = JSON.stringify(global.__coverage__);
+  try {
+    writeFileSync(target, coverageInfo);
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+function setup() {
+  var reallyReallyExit = process.reallyExit;
+
+  process.reallyExit = function(code) {
+    writeCoverage();
+    reallyReallyExit(code);
+  };
```

Comment:
`process.reallyExit` is a C++ function.

> Also, could you point to a test that hits this but not the exit event?

I put this in there to try to be really sure that `writeCoverage()` is called, especially since we have code that tests `process.exit()` itself… I can try to run coverage locally later without this, and see if there’s any difference